blobs.go

```diff
@@ -75,6 +75,17 @@ type BlobDeleter interface {
 	Delete(ctx context.Context, dgst digest.Digest) error
 }
 
+// BlobEnumerator allows to list blobs in storage.
+type BlobEnumerator interface {
+	// Enumerate fills 'digests' with a lexicographically sorted list of
+	// digests up to the size of 'digests' and returns the value 'n' for the
+	// number of entries which were filled. 'last' contains an offset in the
+	// list. It's a string representation of digest. Empty string means 0
+	// offset. 'err' will be set to io.OEF if there are no more entries to
+	// obtain.
+	Enumerate(ctx context.Context, digests []digest.Digest, last string) (n int, err error)
```

Comment:
For posterity, https://github.com/docker/distribution/pull/1050#discussion_r42056309 has me telling you to do something different:

> This allocates way too much memory to serve large requests. Please follow the approach done for Namespace.Repositories so that the dataset can be served in a blocks.

I think a callback based approach addresses this concern.

Sorry I wasn't more specific about this before.